byzcoin/heartbeat.go

```diff
@@ -17,17 +17,18 @@ type heartbeat struct {
 	getTimeChan chan chan time.Time
 	timeout     time.Duration
 	timeoutChan chan string
+	updateTO    chan time.Duration
 }
 
 type heartbeats struct {
 	sync.Mutex
 	wg           sync.WaitGroup
-	heartbeatMap map[string]heartbeat
```

Comment:
What problem does changing this to a pointer solve? I've not got a real opinion one way or the other, but Kelong probably had a reason for not using a pointer there...

Reply:
Reverted to value.

Reply:
Reverted to pointer, so that the timeout update gets stored correctly.